### PR TITLE
PAYARA-2897 Moved generic JASPIC code to provider project

### PR DESCRIPTION
**Pull request by arjantijms**
Refactored client code and improved some naming

### CODE REVIEW

**Comment by arjantijms** [comment]
Jenkins test please

File: appserver/security/jaspic-provider-framework/src/main/java/com/sun/jaspic/services/RegistrationWrapperRemover.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/*
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+ *
+ * Copyright (c) 1997-2012 Oracle and/or its affiliates. All rights reserved.
```
Comment (MattGill98) [inline appserver/security/jaspic-provider-framework/src/main/java/com/sun/jaspic/services/RegistrationWrapperRemover.java]: Newly added classes shouldn't have an oracle header, unless it's been moved from somewhere?

Comment (arjantijms) [reply appserver/security/jaspic-provider-framework/src/main/java/com/sun/jaspic/services/RegistrationWrapperRemover.java]: It's "sort of" moved. The method is extracted from an existing Oracle copyrighted interface into its own interface. I think the argument can go both ways, it's both a moved method and a new class. I meditated over this for a while, but in the end I decided it was more about a method being moved, and less about a class being created from scratch, hence the Oracle header.

Comment (MattGill98) [reply appserver/security/jaspic-provider-framework/src/main/java/com/sun/jaspic/services/RegistrationWrapperRemover.java]: That's fair enough then, I just wanted to make sure it wasn't something forgotten about!

**Pull request merged**